src/Components/ArticleByTopic.jsx

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from "react";
+import getAllArticles from "../api";
+import ArticleCard from "./ArticleCard";
+import { useParams } from "react-router-dom";
+
+const ArticleByTopic = () => {
```

Comment:
All very good 👍 and nice use of useParams :) 

The only thing i would highlight is that, because the article-by-topic and all-articles components have the same layout, they could have been handled together (i.e. only in one component). But this is not an issue that you need to fix as it works well this way too.